tests/integration/models/torch/lighting_modules/test_segmentation.py

```diff
@@ -0,0 +1,163 @@
+import pytest
+import torch
+from pytorch_lightning import LightningModule
+import pytest
+import torch
+import time
+from pytorch_lightning import LightningModule
+from omegaconf import DictConfig
+from tests.fixtures.config import losses as fixt_losses
+from tests.fixtures.config import models as fixt_models
+from tests.fixtures.config import optimizers as fixt_optimizers
+from tests.fixtures.config import trainers as fixt_trainers
+from pytorch_lightning import Trainer
+import pytorch_lightning as pl
+from innofw.core.models.torch.lightning_modules.segmentation import SemanticSegmentationLightningModule
+from innofw.utils.framework import get_model
+from innofw.utils.framework import get_losses
+from torch.utils.data import DataLoader
+from torch.utils.data import Dataset
+from innofw.constants import SegDataKeys
+from innofw.constants import Frameworks
+import tempfile
+import shutil
+import os
+from pytorch_lightning.callbacks import ModelCheckpoint
+
+
+from innofw.core.models.torch.lightning_modules.segmentation import SemanticSegmentationLightningModule
+
+
+# we should try different losses, different optimizsers and different schedulaers and different models
+
+
+
+class DummyDataset(Dataset):
+    def __init__(self, num_samples):
+        self.num_samples = num_samples
+
+    def __getitem__(self, index):
+        x = torch.rand(3, 224, 224)  
+        y = torch.randint(0, 2, (1, 224, 224)) 
+        return {SegDataKeys.image: x, SegDataKeys.label: y}
+
+    def __len__(self):
+        return self.num_samples
+
+
+class DummyDataModule(pl.LightningDataModule):
+    def __init__(self, num_samples: int, batch_size: int = 4):
+        super().__init__()
+        self.num_samples = num_samples
+        self.batch_size = batch_size
+
+    def setup(self, stage= None):
+        self.dataset = DummyDataset(self.num_samples)
+
+    def train_dataloader(self):
+        return DataLoader(self.dataset, batch_size=self.batch_size)
+
+    def val_dataloader(self):
+        return DataLoader(self.dataset, batch_size=self.batch_size)
+
+    def test_dataloader(self):
+        return DataLoader(self.dataset, batch_size=self.batch_size)
+
+@pytest.fixture
+def segmentation_module() -> LightningModule:
```

Comment:
let's put fixture to the fixture folder, or conftest.py file

Reply:
Can you please the path to where can I add it? Just to make sure I will be adding it in the correct place.

Reply:
I would say here: tests/integration/models/torch/lighting_modules/conftest.py

Reply:
fixed